test/web-platform-tests/to-upstream/html/semantics/forms/form-submission-0/submit-event.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>HTML Test: form submit event</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#form-submission-algorithm">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<body>
+<script>
+"use strict";
+
+let submitEvent;
+function prepareForm() {
+  submitEvent = undefined;
+  const form = document.createElement("FORM");
+  const submitButton = document.createElement("BUTTON");
+  form.appendChild(submitButton);
+  form.addEventListener("submit", event => {
+    event.preventDefault();
+    submitEvent = event;
+  });
+  document.body.appendChild(form);
+  return { form, submitButton };
+}
+
+test(() => {
+  const { submitButton } = prepareForm();
+  submitButton.click();
+  assert_equals(submitButton, submitEvent.submitter, "Submitter is set");
+}, "Clicking a submit button sets the submitter");
+
+test(() => {
+  const { form, submitButton } = prepareForm();
+  form.requestSubmit(submitButton);
+  assert_equals(submitButton, submitEvent.submitter, "Submitter is set");
+}, "Requesting submit with a submitter sets the submitter");
+
+test(() => {
+  const { form } = prepareForm();
+  form.requestSubmit();
+  assert_equals(undefined, submitEvent.submitter, "Submitter is not set");
+}, "Requesting submit without a submitter does not set the submitter");
+
+test(() => {
+  const { form } = prepareForm();
+  form.submit();
+  assert_equals(undefined, submitEvent.submitter, "Submitter is not set");
```

Comment:
This does not fire a submit event in browsers, and so this test is invalid. So this means jsdom actually has a problem with its current implementation...

It's not easy to write a test for this since in browsers it will cause a navigation. Hmm.

Reply:
i guess ideally once that bug is fixed, this test should just be removed? i'd be happy to take a stab at fixing that issue in another PR

Reply:
Well, we should not add tests that fail in browsers. And we shouldn't be passing tests that fail in browsers.

Ideally, we should write a test that shows that the event does *not* happen. However, writing such a test is difficult, because in a browser it would cause a navigation.

Reply:
yeah that makes sense, testing this gets a bit tricky 😕

i took a stab at just removing [these lines](https://github.com/jsdom/jsdom/blob/a7c85452d0c4b1947b0410baf11b040f0ed076a9/lib/jsdom/living/nodes/HTMLFormElement-impl.js#L83-L85) on master, and the wpt/tuwpt tests all passed locally. so that's promising, though perhaps there are factors i'm not considering 🤔

Reply:
relevant existing issue: #3435

Reply:
@domenic wdyt of removing the test altogether from this PR (since here we're mainly focused on SubmitEvent.submitter, not whether or not the `submit` event fires)? then we can figure out the best way to test the behavior of `submit()` on #3493

Reply:
Yeah, that sounds perfect. Thanks for working on it!

Reply:
Sure thing, done!